platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblems.kt

```diff
@@ -221,6 +221,7 @@ class ConfigurationCacheProblems(
             }
             .documentationSection(DocumentationSection.TaskOptOut).build()
         report.onIncompatibleTask(problem)
+        summarizer.onIncompatibleTask()
```

Comment:
🤔  With this, we have two sources of truth when it comes to incompatible tasks. Before, it was only the current class, and we used the `incompatibleTasks` map only for reporting the CC end-of-build message.

Should we transfer incompatible task tracking logic to the summarizer as well?

We use the summarizer to guard against duplicate _problems_, we can use it to guard against duplicate incompatible tasks as well. This way, the incompatible task counting logic would be consolidated.